.changeset/blue-eels-invite.md

```diff
@@ -0,0 +1,11 @@
+---
+"fluid-framework": minor
+"@fluidframework/tree": minor
+---
+---
+"section": tree
+---
+
+Better type errors for invalid recursive schema
```

Comment:
Cases which previously compiled but will now fail would have previously failed at runtime, yes? So there is no risk of breaking consumers here?

Reply:
In most cases they would have generally failed at compile time if you actually used the schema with confusing non-local errors. If you didn't use the schema, then I guess you could have new errors.

Reply:
Of if you followed dour documented best practices, would have produced a compile time error in ValidateRecursievSchema.